File: DMOJ/Misc/Balance.py
import sys
from itertools import *
input = sys.stdin.readline

n = int(input())
arr = list(map(int, input().split()))
ttl = sum(arr)
sol = 100000
for i in range(1, n+1):
    for j in list(combinations(arr, i)):
        #print(j, sum(j))
        sol = min(sol, abs(sum(j)*2-ttl))

print(sol)

File: DMOJ/Misc/Balance.cpp
// Balance

#include <bits/stdc++.h>
using namespace std;

#pragma GCC optimize ("Ofast")
#pragma GCC target ("avx2")
#define ms memset
#define pb push_back
#define nl "\n"

int n, arr[10], pre[11], sum = 0, sol = INT_MAX;
int main(){
    ios_base::sync_with_stdio(0);
    cin.tie(nullptr);
    cin >> n;
    for (int i=0; i<n; i++) cin >> arr[i];
    sort(arr, arr+n);
    pre[0] = 0;
    for (int i=0; i<n; i++){
        pre[i+1] = pre[i]+arr[i];
        sum+=arr[i];
    }
    for (int i=0; i<=n; i++)
        sol = min(sol, abs(pre[i]*2-sum));
    cout << sol << nl;

    return 0;
}